pr_agent/algo/ai_handlers/langchain_ai_handler.py

```diff
@@ -71,3 +49,29 @@ async def chat_completion(self, model: str, system: str, user: str, temperature:
         except (Exception) as e:
             get_logger().error("Unknown error during OpenAI inference: ", e)
             raise e
+    
+    @functools.cache
+    def _get_chat(self, deployment_id = None):
```

Comment:
can you explain these lines ?
why do we need cache ? does it actually work in real deployments? 

i would recommend removing the catch line. Seems like a false optimization to me, since I don't see a use case where it would actually work

Reply:
It was a quick way to ensure a single chat object existed per LLM deployment. I assumed chat initialization and first use could have some I/O overhead though thinking more about it I agree I'll remove it.